frontend/catalyst/jax_primitives.py

```diff
@@ -497,11 +504,12 @@ def _apply_registered_pass_lowering(
     transform_mod_type = ir.OpaqueType.get("transform", 'op<"builtin.module">')
     module = jax_ctx.module_context.module
     named_sequence_op = None
-    for op in reversed(module.body.operations):
+    for op in reversed(module.parent.regions[0].blocks[0].operations):
```

Comment:
It has a little to do. It looks like the python bindings for `ModuleOp` / `Module` are a bit weird... `ModuleOp` allows me to give modules a name, while `Module` does not. `Module` has `body` defined while `ModuleOp` does not. I think I can cast one to the other somehow using the bindings, but I was unable to find it. I need modules to have names (i.e., be symbols)

Reply:
Ah! It is not the same.

Since the peep hole optimization attaches itself to qnodes (now functions inside nested modules) and we are interested in finding the transform module (another nested module but nested inside the root module) then what we do is depicted in the following example:

```
#pseudocode
module @root {
  module @nested {
    func.func @qnode
  }
  
  module @transform ...
}
```

Once we are here, it means we are somewhere close to `@qnode` and we want to go to `@transform`. So we get the the current module `@nested`, then we go to its parent `@root`. Then we iterate over its operations to find `@transform`.

Reply:
Ah got it, thanks :)

Reply:
https://github.com/PennyLaneAI/catalyst/pull/1144/commits/f830146a06d902fe482f5344d21cb9071dcdff16